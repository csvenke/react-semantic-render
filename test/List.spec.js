import React from "react";
import renderer from "react-test-renderer";

import { List as List1 } from "../dist/index.cjs";
import { List as List2 } from "../dist/index.esm";
import List3 from "../dist/List";
import List4 from "../src/List";

const describeEach = describe.each([List1, List2, List3, List4]);

const input = [1, 2, 3];

describeEach("with render", (List) => {
  it("should return primary content from render", () => {
    const element = (
      <List items={input} render={(n) => <div key={n}>{n}</div>} />
    );
    const component = renderer.create(element);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("with children", (List) => {
  it("should return primary content from children", () => {
    const element = <List items={input}>{(n) => <div key={n}>{n}</div>}</List>;
    const component = renderer.create(element);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("with children and render", (List) => {
  it("should return primary content from children and ignore render", () => {
    const element = (
      <List items={input} render={(n) => <a key={n}>{n}</a>}>
        {(n) => <div key={n}>{n}</div>}
      </List>
    );
    const component = renderer.create(element);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describeEach("without children and render", (List) => {
  it("should return null", () => {
    const element = <List items={input} />;
    const component = renderer.create(element);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
