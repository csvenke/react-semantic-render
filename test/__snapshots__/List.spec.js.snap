// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with children and render should return primary content from children and ignore render 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children and render should return primary content from children and ignore render 2`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children and render should return primary content from children and ignore render 3`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children and render should return primary content from children and ignore render 4`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children should return primary content from children 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children should return primary content from children 2`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children should return primary content from children 3`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with children should return primary content from children 4`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with render should return primary content from render 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with render should return primary content from render 2`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with render should return primary content from render 3`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`with render should return primary content from render 4`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`without children and render should return null 1`] = `null`;

exports[`without children and render should return null 2`] = `null`;

exports[`without children and render should return null 3`] = `null`;

exports[`without children and render should return null 4`] = `null`;
